src/Altinn.Notifications.Email/Controllers/DeliveryReportController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Text.Json;
+using Altinn.Notifications.Email.Attributes;
+using Altinn.Notifications.Email.Core;
+using Altinn.Notifications.Email.Core.Status;
+
+using Azure.Messaging.EventGrid;
+using Azure.Messaging.EventGrid.SystemEvents;
+
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace Altinn.Notifications.Email.Controllers;
+
+/// <summary>
+/// Controller for handling delivery reports from Azure Communication Services
+/// </summary>
+[Route("notifications/email/api/v1/reports")]
+[ApiController]
+[AccessKey]
+[SwaggerResponse(401, "Caller is unauthorized")]
+public class DeliveryReportController : ControllerBase
+{
+    private readonly IStatusService _statusService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DeliveryReportController"/> class.
+    /// </summary>
+    public DeliveryReportController(IStatusService statusService)
+    {
+        _statusService = statusService;
+    }
+
+    /// <summary>
+    /// Post method for handling delivery reports from Azure Communication Services
+    /// </summary>
+    [HttpPost]
+    [Produces("application/json")]
+    [SwaggerResponse(200, "The delivery report is received")]
+    [SwaggerResponse(400, "The delivery report is invalid")]
+    public async Task<ActionResult<string>> Post([FromBody] EventGridEvent[] eventList)
+    {
+        foreach (EventGridEvent eventgridevent in eventList)
+        {
+            // If the event is a system event, TryGetSystemEventData will return the deserialized system event
+            if (eventgridevent.TryGetSystemEventData(out object systemEvent))
+            {
+                switch (systemEvent)
+                {
+                    case SubscriptionValidationEventData subscriptionValidated:
```

Comment:
would this be a suitable place to comment WHY we return the responsdata ? I guess people can se what is done, but it is not intuitive why?

Reply:
Done